File: src/leetcode/python/two_pointers/MinimumSizeSubarraySum_209.py
class MinimumSizeSubarraySum_209(object):
    def minSubArrayLen(self, s, nums):
        """
        :type s: int
        :type nums: List[int]
        :rtype: int
        """
        import sys
        i, j, subsum = 0, 0, 0
        minlen = sys.maxsize
        while j < len(nums):
            subsum += nums[j]
            j += 1
            while subsum >= s:
                minlen = min(minlen, j - i)
                subsum -= nums[i]
                i += 1
        return minlen if minlen != sys.maxsize else 0

File: src/leetcode/java/MinimumSizeSubarraySum_209.java
package leetcode.java;

public class MinimumSizeSubarraySum_209 {
  public int minSubArrayLen(int s, int[] nums) {
    int minLen = Integer.MAX_VALUE;
    int i = 0, j = 0;
    int sum = 0;
    while (j < nums.length) {
      sum += nums[j++];
      while (sum >= s) {
        minLen = Math.min(minLen, j - i);
        sum -= nums[i++];
      }
    }
    return minLen == Integer.MAX_VALUE ? 0 : minLen;
  }
}
